rts/Lua/LuaSyncedRead.cpp

```diff
@@ -4150,6 +4151,27 @@ int LuaSyncedRead::GetUnitCosts(lua_State* L)
 	lua_pushnumber(L, unit->cost.energy);
 	return 3;
 }
+/***
+ * @function Spring.GetUnitCostTable
+ * @number unitID
+ * @treturn nil|{ metal = number, energy = number }, number buildTime
+ * @treturn number buildTime
+ */
+int LuaSyncedRead::GetUnitCostTable(lua_State* L)
+{
+	const CUnit* const unit = ParseInLosUnit(L, __func__, 1);
+	if (unit == nullptr)
+		return 0;
+	lua_createtable(L, 2, 1);
```

Comment:
The numbers are the array part (for consecutive numerical keys: 1, 2, 3 ...) and the hash part (all other keys). Here we have two keys, "metal" and "energy", which both go into the hash part:
```suggestion
	lua_createtable(L, 0, 2);
```

Reply:
The lua documentation regarding on the lua_createtable function is terrible. Thx.